### PR TITLE
Connect to api

### PR DESCRIPTION
**Pull request by tajemouti**
## Hi Reviewer 👋 ##
Below is a list of the accomplished tasks while working on this milestone:

1. Created a feature branch for this milestone.
2. Integrated Bookstore API.
3. Utilized Axios and createAsyncThunk.
4. Refactored Add and Remove to use API.
5. Implemented Book categories.
6. Updated Redux Thunks and Reducers.
7. Added some basic styling.
8. Fixed linters' errors.

### CODE REVIEW

File: src/components/Book.js [hunk 1]
```diff
@@ -14,23 +14,27 @@ const Book = (props) => {
```
Comment (AbayomiOlaoye) [inline src/components/Book.js:14]: - [ ] You are almost there, @tajemouti :clap: I think you just have to check your `BookPage` again in order to make it conform with the project requirement, and how you can fix the following things:
  - [ ] The name of the component my be better of if it maintains the suggested `<BookList />`
  - [ ] The requirement asked that you fetchBook using a `useEffect` hook to manage the post-event behaviour as shown below :point_down: 
image

**Review by AbayomiOlaoye — CHANGES_REQUESTED** [review]
## STATUS: CHANGE REQUIRED ♻️ 

Hi @tajemouti :wave: , 

Good job with this milestone so far! What a brilliant attempt!!
There are just a few more issues to work on before moving on to the next project but you are almost there!

so_close

## Highlights
- Descriptive Pull Request message :heavy_check_mark: 
- Good use of GitHub flow :heavy_check_mark: 
- Meaningful commit messages :heavy_check_mark: 
- No linter error :heavy_check_mark: 
- Descriptive ReadMe documentation :heavy_check_mark: 
- Components were created cleanly :heavy_check_mark: 

## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Book.js (hunk 1 above)
Comment (tajemouti) [reply src/components/Book.js:14]: Hi @AbayomiOlaoye,
Thank you for taking the time to review my code. I really appreciate it. But I can't see those specific requirements you pointed out in your feedback on the project requirements.
Capture d'écran 2023-09-01 031122
Here's the [link](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connect_to_api_v2_1.md) to this project page.

Comment (AbayomiOlaoye) [reply src/components/Book.js:14]: Oh, yes! Sorry about that but I those files are not among the recent changes you made. The best way to acces that will be to look at your file directory - `BookPage` is in your components folder

Comment (tajemouti) [reply src/components/Book.js:14]: It's alright @AbayomiOlaoye. However, I think that my code aligns with the project requirements, and the changes you required are not listed in the project requirements. The Book component contains the book list element markups.
Cheers

**Review by DammyShittu — APPROVED** [review]
Hi @tajemouti,

## APPROVED 🟢

## Highlights
✔️ Connection to API

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**